osu.Game/Screens/OnlinePlay/Multiplayer/Spectate/CatchUpSpectatorPlayerClock.cs

```diff
@@ -28,9 +28,17 @@ public class CatchUpSpectatorPlayerClock : ISpectatorPlayerClock
 
         public bool IsRunning { get; private set; }
 
+        private bool isReady;
+
         public void Reset() => CurrentTime = 0;
 
-        public void Start() => IsRunning = true;
+        public void Start()
+        {
+            if (!isReady)
+                throw new InvalidOperationException(@"Attempting to start spectator clock while not ready for use yet.");
```

Comment:
Really unsure about this being an exception..

Reply:
Yeah, have mentioned in OP:
> The guard may be harsh on users, and could potentially downgrade to an Important log that's at least reported to sentry or something? not sure on the direction forward regarding that.

Meanwhile if this ever hits, it means something is deeply wrong with our current gameplay structure and multi-spectator is very likely to break again. We can consider switching to `Debug.Assert` to limit this to only us for noticing it?

You could alternatively look at the [other suggestion](https://github.com/ppy/osu/pull/18245#discussion_r871819457) which removes this guard and basically never allow this to possibly happen in the first place.

Reply:
I'm still figuring out the over-arching fix here, so not ready to know which solution I prefer. We'll see.